packages/pg-pool/index.js

```diff
@@ -66,11 +66,17 @@ class Pool extends EventEmitter {
     if (options != null && 'password' in options) {
       // "hiding" the password so it doesn't show up in stack traces
       // or if the client is console.logged
+      this.options.password = options.password
       Object.defineProperty(this.options, 'password', {
-        configurable: true,
         enumerable: false,
-        writable: true,
-        value: options.password,
+      })
+    }
+    if (options != null && options.ssl && options.ssl.key) {
+      // "hiding" the ssl->key so it doesn't show up in stack traces
+      // or if the client is console.logged
+      this.options.ssl.key = options.ssl.key
```

Comment:
This was done with `password` because `Object.assign` doesn’t include it when it’s non-enumerable, but the same doesn’t apply to `key` because it’s nested in `ssl`. (At the same time, subtly modifying the input might not be a great idea. Maybe we should just make the whole `ssl` key non-enumerable? Or the client non-enumerable on errors so there’s no need to worry about what’s on the client and when various ways to read options will and won’t work?)

Reply:
The whole SSL cannot be non enumerable. Have you tried what you are suggesting? Try making cert for example non-enumerable.

Reply:
Sorry, let me clarify: `this.options.ssl.key = options.ssl.key` accomplishes nothing, because `this.options.ssl === options.ssl` in this branch. The value for `password` was set explicitly because `Object.assign({}, options)` doesn’t copy non-enumerable properties, but there’s no `options.tls = Object.assign({}, options.tls)`.

Reply:
the object.assign will miss if config is non-enumerable though right? Right now people who don't want to leak their key (me) are sending in a non-enumerable key

Minimal reproducable use-case

```
// works because enumerable by defaul
Object.assign({}, {password: 'bob'})
08:03:52.471 {password: "bob"}

a = {password: 'bob'}
Object.defineProperty(a, 'password', {
    enumerable: false
})
// wait for it
Object.assign({}, a)
08:05:39.253 {}
```

Reply:
That example is why we do it for `password`, yes. `tls.key` is different:

```js
let a = {password: 'bob', ssl: {key: 'bob'}};
Object.defineProperty(a, 'password', {enumerable: false});
Object.defineProperty(a.ssl, 'key', {enumerable: false});

Object.assign({}, a).ssl.key
```

Because the `ssl` property is copied, but not the object it contains – i.e. `this.options.ssl === options.ssl` – this line just assigns a property back to itself, which does nothing. It can be removed without changing the overall behaviour\*.

```suggestion
```

This also means that the `Object.defineProperty` alters something passed into the constructor with subtle effects on the outside, so maybe we *should* copy `ssl`.

\* except when `key` comes from the prototype chain of `ssl` somehow/has a setter/is on a proxy, in which case we still don’t want to do this.

Reply:
Let's try it...

Reply:
Works 👍  I'll try to re-read later after work.